File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int n=nums.size();
        if(n==0) return 0;

        sort(nums.begin(),nums.end());

        int cnt=0;
        int l=1;
        int ls=INT_MIN;
        for(int i=0;i<n;i++){
            if(nums[i]-1==ls){
                cnt++;
                ls=nums[i];
        }
        else if(nums[i]!=ls){
            cnt=1;
            ls=nums[i];

        }
        l=max(l,cnt);
        }
        return l;
        
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        n=len(nums)
        l,r=0,0
        if not nums:
            return 0
        nums.sort()
        length=0
        common=0
        while r<n-1:
            if (nums[r]+1==nums[r+1]):
                r+=1
            elif(nums[r]==nums[r+1]):
                common+=1
                r+=1
            else:
                length=max(length,r-l+1-common)
                l=r+1
                r+=1
                common=0
        length=max(length,r-l+1-common)
        return length